From: Sam James <sam@gentoo.org>
Date: Mon, 12 Sep 2022 18:02:21 +0100
Subject: configure.ac: fix configure tests broken with Clang 15 (implicit
 function declarations)

Clang 15 makes implicit function declarations fatal by default which
leads to some configure tests silently failing/returning
the wrong result.

Signed-off-by: Sam James <sam@gentoo.org>
(cherry-picked from commit d0a3570300812bc81888e7a7eadb9311621dc9a7)
Origin: upstream, 2.24.1, commit:0680314c2aa48997df67bb4ae51be017bc9ef643
---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index 71d51a0..1295d39 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2725,6 +2725,7 @@ CheckInputKD()
     AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
       #include <linux/kd.h>
       #include <linux/keyboard.h>
+      #include <sys/ioctl.h>
     ]], [[
         struct kbentry kbe;
         kbe.kb_table = KG_CTRL;
